src/server.cpp

```diff
@@ -1032,11 +1032,6 @@ PlayerSAO* Server::StageTwoClientInit(session_t peer_id)
 	// Send Breath
 	SendPlayerBreath(playersao);
 
-	// Note things in chat if not in simple singleplayer mode
-	if (!m_simple_singleplayer_mode && g_settings->getBool("show_statusline_on_connect")) {
-		// Send information about server to player in chat
-		SendChatMessage(peer_id, ChatMessage(CHATMESSAGE_TYPE_SYSTEM, getStatusString()));
```

Comment:
does this make `getStatusString()` dead code?

Reply:
No, it's still used (in one line) for `core.get_server_status()`.

Reply:
I know but `CHATMESSAGE_TYPE_RAW` (from `core.chat_send_player`) and `CHATMESSAGE_TYPE_SYSTEM` are handled exactly the same way in `Client::getChatMessage`.
These different types don't seem to be used anywhere outside the previously mentioned function. Is the purpose for them still in progress or is it legacy left-over?

Reply:
it was introduced with the chat rework, intended to CSM to know which type is it and handle it as it wants, like adding colors for each type, for example

Reply:
Sounds good, but this will definitely need undoing the 3rd argument for `chat_send_player`, so the chat message type can be specified by Lua (prepend removed in 9a3b771).
After all also the player join message is [sent with that function](https://github.com/minetest/minetest/blob/master/builtin/game/misc.lua#L48), which should actually be marked as a server message too.

Reply:
yes it's exact